libnopegl/src/text.h

```diff
@@ -89,14 +98,34 @@ struct text_cls {
     uint32_t flags; // combination of NGLI_TEXT_FLAG_*
 };
 
+/* Each field points to a contiguous data buffer (1 row per character) */
+struct text_effects_pointers {
+    float *transform; // mat4[]
+    float *color;     // vec3[]
+    float *opacity;   // f32[]
+};
+
+struct effect_segmentation {
+    size_t *positions;     // character index (in chars darray) to position in "target unit" (char, word, ...)
+    size_t total_segments; // total number of segment: all values in positions are between [0;total_segments-1]
+};
+
 struct text {
     struct ngl_ctx *ctx;
     struct text_config config;
 
+    /* user exposed */
```

Comment:
public ?

Reply:
done (I wanted to avoid the term "public" because I associate it with `nopegl.h`)